rasa/core/domain.py

```diff
@@ -264,31 +268,68 @@ def collect_slots(slot_dict: Dict[Text, Any]) -> List[Slot]:
+        properties.setdefault(IGNORE_ENTITIES_KEY, [])
+        if not properties[USE_ENTITIES_KEY]:  # this covers False, None and []
+            properties[USE_ENTITIES_KEY] = []
+
+        # `use_entities` is either a list of explicitly included entities
+        # or `True` if all should be included
+        if properties[USE_ENTITIES_KEY] is True:
+            included_entities = set(entities)
+        else:
+            included_entities = set(properties[USE_ENTITIES_KEY])
+        excluded_entities = set(properties[IGNORE_ENTITIES_KEY])
+        used_entities = list(included_entities - excluded_entities)
+        used_entities.sort()
+
+        # Only print warning for ambiguous configurations if entities were included
+        # explicitly.
+        explicitly_included = isinstance(properties[USE_ENTITIES_KEY], list)
+        ambiguous_entities = included_entities.intersection(excluded_entities)
+        if explicitly_included and ambiguous_entities:
+            raise_warning(
+                f"Entities: '{ambiguous_entities}' are explicitly included and"
+                f" excluded for intent '{name}'."
+                f"Excluding takes precedence in this case. "
+                f"Please resolve that ambiguity.",
+                docs=f"{DOCS_URL_DOMAINS}#ignoring-entities-for-certain-intents",
+            )
+
+        properties[USED_ENTITIES_KEY] = used_entities
+        properties.pop(USE_ENTITIES_KEY)
+        properties.pop(IGNORE_ENTITIES_KEY)
+
+        return intent
+
+    @classmethod
     def collect_intent_properties(
-        intents: List[Union[Text, Dict[Text, Any]]]
+        cls, intents: List[Union[Text, Dict[Text, Any]]], entities: List[Text]
     ) -> Dict[Text, Dict[Text, Union[bool, List]]]:
+
         intent_properties = {}
         for intent in intents:
-            if isinstance(intent, dict):
-                name = list(intent.keys())[0]
-                for properties in intent.values():
-                    properties.setdefault("use_entities", True)
-                    properties.setdefault("ignore_entities", [])
-                    if (
-                        properties["use_entities"] is None
-                        or properties["use_entities"] is False
-                    ):
-                        properties["use_entities"] = []
-            else:
-                name = intent
-                intent = {intent: {"use_entities": True, "ignore_entities": []}}
+            if not isinstance(intent, dict):
+                intent = {intent: {USE_ENTITIES_KEY: True, IGNORE_ENTITIES_KEY: []}}
 
+            name = list(intent.keys())[0]
             if name in intent_properties.keys():
                 raise InvalidDomain(
-                    "Intents are not unique! Found two intents with name '{}'. "
-                    "Either rename or remove one of them.".format(name)
+                    f"Intents are not unique! Found two intents with name '{name}'. "
```

Comment:
```suggestion
                    f"Intents are not unique! Found multiple intents with name '{name}'. "
```
there may be more than two, right?

Reply:
might it make sense to first collect all duplicate entities in the loop, then raise the exception afterwards if there are any? that way we can give out a list of which ones are duplicate